File: app/src/main/java/nl/rsdt/japp/jotial/maps/management/MapItemDateControl.java
package nl.rsdt.japp.jotial.maps.management;

import java.util.Calendar;

import nl.rsdt.japp.jotial.Destroyable;

/**
 * @author Dingenis Sieger Sinke
 * @version 1.0
 * @since 31-7-2016
 * Description...
 */
public abstract class MapItemDateControl implements Destroyable {

    protected Calendar lastUpdate = Calendar.getInstance();

    public Calendar getLastUpdateDate() {
        return lastUpdate;
    }

    protected boolean isElapsedSinceLastUpdate(long time) {
        Calendar now = Calendar.getInstance();
        long delta = now.getTimeInMillis() - lastUpdate.getTimeInMillis();
        return delta >= time;
    }

    @Override
    public void onDestroy() {
        lastUpdate = null;
    }
}

File: app/src/main/java/nl/rsdt/japp/jotial/maps/management/MapItemDateControl.kt
package nl.rsdt.japp.jotial.maps.management

import nl.rsdt.japp.jotial.Destroyable
import java.util.*

/**
 * @author Dingenis Sieger Sinke
 * @version 1.0
 * @since 31-7-2016
 * Description...
 */
abstract class MapItemDateControl : Destroyable {

    var lastUpdateDate: Calendar? = Calendar.getInstance()
        protected set

    protected fun isElapsedSinceLastUpdate(time: Long): Boolean {
        val now = Calendar.getInstance()
        val delta = now.timeInMillis - lastUpdateDate!!.timeInMillis
        return delta >= time
    }

    override fun onDestroy() {
        lastUpdateDate = null
    }
}
